Yoast/wordpress-seo
# Update grunt task and dependencies

## Summary

This PR can be summarized in the following changelog entry:

* None needed.

## Relevant technical choices:

In general, two things are happening in this pull:
* grunt dependencies are moved from this project to `@yoast/grunt-plugin-tasks`, so we only have to maintain them in one place. 
* CSS generation & minification is handled differently: 
  * ~~Minification is now handled in the `postcss` task by `cssnano`. Because of this, we no longer have `.min.css` files, and only `.css` files, as in both development builds and production builds we work with the same minified files.~~
  * In development builds we generate [CSS source maps](https://www.richfinelli.com/what-is-a-css-source-maps/) for our CSS files. This way our normal CSS files _and_ our RTL CSS files have source maps.
* To differentiate between development builds and production/release builds, we now have a `global.developmentBuild` variable. This variable gets set to `false` when the grunt command you run is one of: `[ 'release', 'release:css', 'release:js', 'artifact', 'deploy:trunk', 'deploy:master' ]`.
* Removed select2 minified CSS file from the repo. The RTL grunt task was adding CSS here and GIT did not ignore it.

## Test instructions

This PR can be tested by following these steps:

* All build tasks should work as normal, please test `build:css`, `release`, `artifact`, `deploy:trunk`, `deploy:master`. When running `build:css`, source maps should be created. When running `release`, source maps should _not_ be created.
	* Run `composer install --no-dev` after `grunt release` to make it work.
* All CSS should be loaded properly where needed, the paths have changed so please test this too.
* Please test whether the CSS source maps work.
* Test RTL CSS as well. Download RTL Tester by `Yoav Farhi, Automattic` to get a very handy language direction switcher in the toolbar.

## Quality assurance
